web/src/components/projects/lib/projects.ts

```diff
@@ -0,0 +1,73 @@
+import { ProjectTreRoleName } from "@/openapi";
+import { ProjectNameValidation, AnyProjectRoleName } from "@/types/projects";
+
+// Project name validation patterns per environment
+// Note: These patterns are also validated on the backend in internal/validation/patterns.go
+const TRE_PROJECT_NAME_PATTERN = /^[a-z0-9][a-z0-9-]*[a-z0-9]$/;
+const TRE_PROJECT_NAME_MIN_LENGTH = 3;
+const TRE_PROJECT_NAME_MAX_LENGTH = 50;
+
+export const getProjectNameValidation = (environmentName: string): ProjectNameValidation => {
```

Comment:
feels quite map like? are JS maps bad?

Reply:
I don't mind really. Something like that: [tweaking frontend validation data structures](https://github.com/ucl-arc-tre/portal/pull/400/commits/83fff607139017c81c8cc561ae82ac97cdd764f8) ?